src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.admin.indices.analyze.AnalyzeAction;
+import org.opensearch.index.analysis.AnalysisRegistry;
+import static org.opensearch.action.admin.indices.analyze.TransportAnalyzeAction.analyze;
+
+/**
+ * The implementation {@link FieldChunker} for fixed token length algorithm.
+ */
+@Log4j2
+public class FixedTokenLengthChunker implements FieldChunker {
+
+    public static final String TOKEN_LIMIT_FIELD = "token_limit";
+    public static final String OVERLAP_RATE_FIELD = "overlap_rate";
+    public static final String MAX_TOKEN_COUNT_FIELD = "max_token_count";
+    public static final String TOKENIZER_FIELD = "tokenizer";
+
+    // default values for each parameter
+    private static final int DEFAULT_TOKEN_LIMIT = 500;
+    private static final BigDecimal DEFAULT_OVERLAP_RATE = new BigDecimal("0");
+    private static final int DEFAULT_MAX_TOKEN_COUNT = 10000;
+    private static final String DEFAULT_TOKENIZER = "standard";
+
+    private static final BigDecimal OVERLAP_RATE_UPPER_BOUND = new BigDecimal("0.5");
+
+    private final AnalysisRegistry analysisRegistry;
+
+    public FixedTokenLengthChunker(AnalysisRegistry analysisRegistry) {
+        this.analysisRegistry = analysisRegistry;
+    }
+
+    private void validatePositiveIntegerParameter(Map<String, Object> parameters, String fieldName) {
+        // this method validate that parameter is a positive integer
+        if (!parameters.containsKey(fieldName)) {
+            // all parameters are optional
+            return;
+        }
+        if (!(parameters.get(fieldName) instanceof Number)) {
```

Comment:
Since you're validating integer, why not check Integer directly? And avoid the force cast to the value to Number.

Reply:
Already changed to numberutil.